### PR TITLE
Add WDL metadata docs

### PR DESCRIPTION
**Pull request by garyluu**
For issue #1490 
Add some docs for WDL metadata

### CODE REVIEW

File: _docs/best-practices.md [hunk 1]
```diff
@@ -185,6 +185,8 @@ s:dateCreated: "2016-12-13"
 s:license: https://www.apache.org/licenses/LICENSE-2.0
 ```
 
+For WDL descriptors, see the [WDL documentation](https://software.broadinstitute.org/wdl/documentation/spec#metadata-section) for how to define metadata.  See [this](https://github.com/ga4gh/dockstore/blob/1.5.0-alpha.4/dockstore-webservice/src/test/resources/metadata_example2.wdl#L194) for an example which defines author, email, and description metadata.
```
Comment (denis-yuen) [inline _docs/best-practices.md]: Please write out a full working example here using a more formal name for the author, email, etc. 
Take inspiration from the above examples.

**Comment by garyluu** [comment]
Removed the previous example, took example from WDL spec and gave it more legit looking description, author, and email

File: _docs/best-practices.md [hunk 2]
```diff
@@ -185,6 +185,38 @@ s:dateCreated: "2016-12-13"
...
+    docker: "broadinstitute/baseimg"
+  }
+  parameter_meta {
+    memory_mb: "Amount of memory to allocate to the JVM"
+    param: "Some arbitrary parameter"
+    sample_id: "The ID of the sample in format foo_bar_baz"
+  }
+  meta {
+    author: "Denis Yuen"
+    email: "dyuen@oicr.on.ca"
```
Comment (denis-yuen) [inline _docs/best-practices.md]: oh right, we're using live emails in this tutorial. 
Replace the brian emails to "@not-ucsc.org" and mine to "@not-oicr.ca"

File: _docs/best-practices.md [hunk 3]
```diff
@@ -185,6 +185,38 @@ s:dateCreated: "2016-12-13"
...
+    description: "An example tool demonstrating metadata. Note that this is an example and the metadata is not necessarily consistent."
+  }
+}
```
[the remark this answers is not in the document]
Comment (garyluu) [reply _docs/best-practices.md]: seems to be valid using womtool without a full workflow...once i remove the duplicate sample_id parameter

**Pull request merged**